crates/neon/src/thread/mod.rs

```diff
@@ -0,0 +1,232 @@
+//! Thread-local storage for JavaScript threads.
+//!
+//! At runtime, an instance of a Node.js addon can contain its own local storage,
+//! which can then be shared and accessed as needed between Rust modules. This can
+//! be useful for setting up long-lived state that needs to be shared between calls
+//! of an addon's APIs.
+//!
+//! For example, an addon may wish to track the [thread ID][threadId] of each of its
+//! instances:
+//!
+//! ```
+//! # use neon::prelude::*;
+//! # use neon::thread::LocalKey;
+//! static THREAD_ID: LocalKey<u32> = LocalKey::new();
+//!
+//! pub fn thread_id<'cx, C: Context<'cx>>(cx: &mut C) -> NeonResult<u32> {
+//!     THREAD_ID.get_or_try_init(cx, |cx| {
+//!         let global = cx.global();
+//!         let require: Handle<JsFunction> = global.get(cx, "require")?;
+//!         let worker: Handle<JsObject> = require.call_with(cx)
+//!             .arg(cx.string("node:worker_threads"))
+//!             .apply(cx)?;
+//!         let threadId: Handle<JsNumber> = worker.get(cx, "threadId")?;
+//!         Ok(threadId.value(cx) as u32)
```

Comment:
Nit, not idiomatic Rust naming.

```suggestion
//!         let thread_id: Handle<JsNumber> = worker.get(cx, "threadId")?;
//!         Ok(thread_id.value(cx) as u32)
```